rasa/train_chunks.py

```diff
@@ -0,0 +1,578 @@
+
+from rasa.shared.utils.cli import print_success, print_warning, print_error, print_color
+import rasa.shared.utils.io
+from rasa.shared.constants import (
+    DEFAULT_MODELS_PATH,
+    DEFAULT_CORE_SUBDIRECTORY_NAME,
+    DEFAULT_NLU_SUBDIRECTORY_NAME,
+)
+
+
+def train_in_chunks(
+    domain: Text,
+    config: Text,
+    training_files: Union[Text, List[Text]],
+    output: Text = DEFAULT_MODELS_PATH,
+    force_training: bool = False,
+    fixed_model_name: Optional[Text] = None,
+    core_additional_arguments: Optional[Dict] = None,
+    nlu_additional_arguments: Optional[Dict] = None,
+    loop: Optional[asyncio.AbstractEventLoop] = None,
+) -> Optional[Text]:
+    """Trains a Rasa model using smaller chunks.
+
+    Args:
+        domain: Path to the domain file.
+        config: Path to the config for Core and NLU.
+        training_files: Paths to the training data for Core and NLU.
+        output: Output path.
+        force_training: If `True` retrain model even if data has not changed.
+        fixed_model_name: Name of model to be stored.
+        core_additional_arguments: Additional training parameters for core training.
+        nlu_additional_arguments: Additional training parameters forwarded to training
+                                  method of each NLU component.
+
+    Returns:
+        Path of the trained model archive.
+    """
+    return rasa.utils.common.run_in_loop(
+        train_in_chunks_async(
+            domain,
+            config,
+            training_files,
+            output,
+            force_training,
+            fixed_model_name,
+            core_additional_arguments,
+            nlu_additional_arguments,
+        ),
+        loop,
+    )
+
+
+async def train_in_chunks_async(
+    domain: Union[Domain, Text],
+    config: Text,
+    training_files: Optional[Union[Text, List[Text]]],
+    output: Text = DEFAULT_MODELS_PATH,
+    force_training: bool = False,
+    fixed_model_name: Optional[Text] = None,
+    core_additional_arguments: Optional[Dict] = None,
+    nlu_additional_arguments: Optional[Dict] = None,
+) -> Optional[Text]:
+    """Trains a Rasa model using smaller chunks (Core and NLU).
+
+    Args:
+        domain: Path to the domain file.
+        config: Path to the config for Core and NLU.
+        training_files: Paths to the training data for Core and NLU.
+        output: Output path.
+        force_training: If `True` retrain model even if data has not changed.
+        fixed_model_name: Name of model to be stored.
+        core_additional_arguments: Additional training parameters for core training.
+        nlu_additional_arguments: Additional training parameters forwarded to training
+                                  method of each NLU component.
+
+    Returns:
+        Path of the trained model archive.
+    """
+    file_importer = TrainingDataImporter.load_from_config(
+        config, domain, training_files
+    )
+    with ExitStack() as stack:
```

Comment:
we can use the `TempDirectoryPath` context manager directly as it's the only context manager here.